_healthcare/views.py

```diff
@@ -13,6 +13,11 @@
 from utils.fhirclient import client
 from utils.fhirclient.models.medicationrequest import MedicationRequest
 from utils.fhirclient.models.claim import Claim
+import uuid
+import cachetools
+
+# Server side cache for storing our FHIR tokens (to avoid making our session too big).
```

Comment:
@dancinnamon-okta Never worked with cachetools but have a little concern when this on the ElastBeanStalk if cached is not looked at, was just looking at the documentation about cachetools and they do have `lock` concepts. Just wondering about caching and multi-tenancy since that would cause some weird behaviors.

Reply:
@dancinnamon-okta Plus Zartan implement a clear session and wondering if we need to account for this. 

Here is the implementation:
https://github.com/udplabs/zartan/blob/2964276c62df47c34a5bf73d2e4af78f22db34c1/utils/udp.py#L53-L63

Reply:
@noinarisak - On the cachetools multi-threaded front, I can take a look at that.  Right now every user has a randomly generated key for their item in the cache, so updates are not to the same item, but admittedly to the same dictionary behind the scenes.

For session- I think we're ok there.  I am storing a pointer (a guid basically) in the actual session that tells me where in the cache to find my additional state.  If that session variable goes away, the item in the cache is no longer accessible, and the TTL will clear it out within 5 minutes or so.